### PR TITLE
Allow for subclassing of master controller Main

### PR DESCRIPTION
**Pull request by jowerner**
- made methods of master controller Main non-static and protected so we can inherit from Main and overwrite these methods as needed
- moved any System.exit() to the new run() method so the JVM does not quit by accident when subclasses call any of the other methods
- adjusted test case

### CODE REVIEW

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java [hunk 1]
```diff
@@ -163,7 +163,7 @@ private static File getOverridePropertieFile(final CommandLine commandLine)
      * @param config
      *            master controller configuration
      */
-    private static void setupHttpsProxy(final MasterControllerConfiguration config)
+    private void setupHttpsProxy(final MasterControllerConfiguration config)
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/mastercontroller/Main.java:166]: Shouldn't this be `protected` to allow overrides?

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java [hunk 2]
```diff
@@ -183,7 +183,7 @@ private static void setupHttpsProxy(final MasterControllerConfiguration config)
      *            master controller configuration
      * @return the HessianProxyFactory
      */
-    private static HessianProxyFactory getHessianProxyFactory(final MasterControllerConfiguration config)
+    private HessianProxyFactory getHessianProxyFactory(final MasterControllerConfiguration config)
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/mastercontroller/Main.java:186]: Shouldn't this be protected to allow overrides?

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java [hunk 3]
```diff
@@ -201,7 +201,7 @@ private static HessianProxyFactory getHessianProxyFactory(final MasterController
      *            master controller configuration
      * @return the UrlConnectionFactory
      */
-    private static UrlConnectionFactory getUrlConnectionFactory(final MasterControllerConfiguration config)
+    private UrlConnectionFactory getUrlConnectionFactory(final MasterControllerConfiguration config)
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/mastercontroller/Main.java:204]: Shouldn't this be protected to allow overrides?

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java [hunk 4]
```diff
@@ -75,12 +75,12 @@
 
     private static final Log log = LogFactory.getLog(Main.class);
 
-    private static BasicConsoleUI ui;
+    private BasicConsoleUI ui;
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/mastercontroller/Main.java:78]: Allow subclasses to access this field (w/o the need to override `setupUi`)?

**Review by h-arlt — APPROVED** [review]
Some methods - and the `ui` field - might also be declared `protected` but not necessarily.

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java (hunk 1 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/mastercontroller/Main.java]: Done.

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java (hunk 2 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/mastercontroller/Main.java]: Done.

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java (hunk 3 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/mastercontroller/Main.java]: Done.

File: src/main/java/com/xceptance/xlt/mastercontroller/Main.java (hunk 4 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/mastercontroller/Main.java]: Done.

**Pull request merged**